src/openzaak/components/zaken/tests/test_zaak_filter.py

```diff
@@ -304,3 +304,77 @@ def test_filter_zaaktype_omschrijving(self):
         self.assertEqual(
             response.json()["results"][0]["url"], f"http://testserver{reverse(zaak)}"
         )
+
+    def test_filter_kenmerken_bron(self):
+        kenmerk = ZaakKenmerkFactory(bron="BAG-id")
+        ZaakKenmerkFactory(bron="GAB-id")
+
+        ZaakFactory.create()
+
+        response = self.client.get(
+            self.url,
+            {"kenmerk__bron": "BAG-id"},
+            **ZAAK_WRITE_KWARGS,
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertEqual(response.json()["count"], 1)
+        self.assertEqual(
+            response.json()["results"][0]["url"],
+            f"http://testserver{reverse(kenmerk.zaak)}",
+        )
+
+    def test_filter_kenmerken_bron_duplicate_bronnen(self):
+        zaak = ZaakFactory.create()
+        ZaakKenmerkFactory(bron="BAG-id", zaak=zaak)
+        ZaakKenmerkFactory(bron="BAG-id", zaak=zaak)
+
+        ZaakFactory.create()
+
+        response = self.client.get(
+            self.url,
+            {"kenmerk__bron": "BAG-id"},
+            **ZAAK_WRITE_KWARGS,
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertEqual(response.json()["count"], 1)
+        self.assertEqual(
+            response.json()["results"][0]["url"], f"http://testserver{reverse(zaak)}"
+        )
+
+    def test_filter_kenmerken(self):
+        zaak = ZaakFactory.create()
+        ZaakKenmerkFactory(bron="BAG-id", kenmerk="123", zaak=zaak)
+        ZaakKenmerkFactory(bron="GAB-id", kenmerk="456", zaak=zaak)
+
+        ZaakFactory.create()
+
+        with self.subTest("existing combination"):
```

Comment:
could you also add a test to verify that this works if you explicitly URLencode the query param? E.g. `?kenmerk=foo%3Abar` should also work, `:` should technically be encoded because it's a reserved character in URLs

Reply:
added a test, but it seems that the other tests do encode. `response.request["QUERY_STRING"]` has `%3A`